adder/local/dag_service.go

```diff
@@ -62,7 +66,7 @@ func (dgs *DAGService) Add(ctx context.Context, node ipld.Node) error {
 		CumSize: size,
 	}
 
-	return adder.PutBlock(ctx, dgs.rpcClient, nodeSerial, dgs.dests)
+	return dgs.bph.PutBlock(ctx, nodeSerial)
```

Comment:
NodeWithMeta let's pass `format`

BlockPut uses `v0`, but PutDAG uses `cbor` (v1) (why?)

 I will check if passing format makes things cleaners, but changing NodeWithMeta to IPLD Node will lead to quite a few changes including breaking rpc api for BlockPut

Reply:
Format tells ipfs what codec to give to the CID of the block. Depends on whether we are sending a block from the dag builder (v0 protobufs) or a clusterDAGnode which we use cbor for. I believe, format can also be a field in the helper, since it's fixed for all the things that helper needs to put?

Reply:
> I believe, format can also be a field in the helper, since it's fixed for all the things that helper needs to put?

No. 

bph is part of shard object which uses protobuf on PutBlock and cbor on PutDAG. 

We can make format part of bph and use a new bph everytime PutDAG is called.

Reply:
Or check whether the node passed is cbor `_, ok := n.(*cbor.Node)`

Reply:
whatever you prefer